### PR TITLE
Add log critical to FLComponent and update docstrings

### LINKED ISSUE #53: Add log_critical to FLComponent
We have log_info, log_warning, log_error, log_exception, log_debug functions already.

Add log_critical to be consistent with Python logger.

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
Fixes #53 

- Add log_critical to FLComponent
- Add docstrings to FLComponent

### CODE REVIEW

**Review by yhwen — APPROVED** [review]
looks good.

**Review by yanchengnv — CHANGES_REQUESTED** [review]
Please set the default value of "fire_event" as follows:

True for log_error(), log_warning(), and log_critical()
False for log_debug(), log_info() and log_exception()

Otherwise there will be too many events.

File: nvflare/apis/fl_component.py [hunk 1]
```diff
@@ -52,6 +58,14 @@ def fire_event(self, event_type: str, fl_ctx: FLContext):
         self._fire(event_type, fl_ctx)
 
     def fire_fed_event(self, event_type: str, event_data: Shareable, fl_ctx: FLContext, targets=None):
+        """Fires a federation event.
```
Comment (madil90) [inline nvflare/apis/fl_component.py:61]: should we describe what a federation event is here?

**Review by madil90 — COMMENTED** [review]
Docstrings can be more descriptive. This will become basis of our public documentation.

File: nvflare/apis/fl_component.py [hunk 2]
```diff
@@ -64,46 +78,43 @@ def fire_fed_event(self, event_type: str, event_data: Shareable, fl_ctx: FLConte
...
+            fl_ctx (FLContext): FLContext information.
         """
         pass
 
-    def make_log_message(self, fl_ctx: FLContext, msg: str):
-        return generate_log_message(fl_ctx, msg)
-
     def log_info(self, fl_ctx: FLContext, msg: str, fire_event=True):
-        log_msg = self.make_log_message(fl_ctx, msg)
+        """Logs a message with logger.info.
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/apis/fl_component.py:115]: Updated docstrings

[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/apis/fl_component.py:115]: Thanks for the feedback, I have updated the docstrings

File: nvflare/apis/fl_component.py (hunk 1 above)
Comment (YuanTingHsieh) [reply nvflare/apis/fl_component.py:61]: Thanks for the feedback, I have updated the docstrings

**Comment by YuanTingHsieh** [comment]
The concerns from review are addressed.

**Review by yanchengnv — APPROVED** [review]
Looks good!

**Review by yhwen — APPROVED** [review]
LGTM

**Pull request merged**